coreneuron/apps/main1.cpp

```diff
@@ -518,7 +523,14 @@ extern "C" int run_solve_core(int argc, char** argv) {
         double min_report_dt = INT_MAX;
         int report_buffer_size = nrnopt_get_int("--report-buffer-size");
         for (size_t i = 0; i < configs.size(); i++) {
-            register_report(dt, tstop, delay, configs[i]);
+            std::unique_ptr<ReportHandler> report_handler;
+            if (std::strcmp(configs[i].format, "Bin") == 0) {
+                report_handler = std::make_unique<BinaryReportHandler>(configs[i]);
+            } else if (std::strcmp(configs[i].format, "SONATA") == 0) {
+                report_handler = std::make_unique<SonataReportHandler>(configs[i]);
+            }
```

Comment:
An `else` clause is missing. It should report an error that format is neither "Bin"  nor "SONATA".
Besides, I would move the creation of a `ReportHandler` from a `ReportConfiguration` in a dedicated function.